lib/mongo/bulk_write.rb

```diff
@@ -313,5 +314,45 @@ def can_hint?(connection)
         end
       end
     end
+
+    # Checks the request documents to make sure they are valid. For update
```

Comment:
This description isn't matching the implementation, because the documents are only partially checked. A more accurate description would be something like:

> Perform the request document validation required by driver specifications. This method validates the first key of each update request document to be an operator (i.e. start with `$`) and the first key of each replacement document to not be an operator (i.e. not start with `$`). The request document may be invalid without this method flagging it as such (for example an update or replacement document containing some keys which are operators and some which are not), in which case the driver expects the server to fail the operation with an error.

Reply:
done